### PR TITLE
server: add metric for conn send/recv bytes.

### PR DESCRIPTION
**Pull request by siddontang**
@BusyJay @hhkbp2 @javaforfun

### CODE REVIEW

File: src/server/conn.rs [hunk 1]
```diff
@@ -271,7 +277,9 @@ impl Conn {
         where T: RaftStoreRouter,
               S: StoreAddrResolver
     {
+        let n = self.send_buffer.len();
         msg.encode_to(&mut self.send_buffer).unwrap();
+        CONN_PENDING_SEND_BYTES_COUNTER.inc_by((self.send_buffer.len() - n) as f64).unwrap();
```
Comment (andelf) [inline src/server/conn.rs]: is it a counter or gauge?

Comment (siddontang) [inline src/server/conn.rs]: counter

**Comment by javaforfun** [comment]
LGTM

**Comment by zhangjinpeng1987** [comment]
LGTM

File: src/server/conn.rs (hunk 1 above)
Comment (BusyJay) [inline src/server/conn.rs]: Doesn't it need to be decreased when the buffer is consumed?

Comment (siddontang) [inline src/server/conn.rs]: No, here we will calculate the rate of pending bytes in Prometheus, it will be increased all time.

Comment (BusyJay) [inline src/server/conn.rs]: Then the name is confusing.

Comment (siddontang) [inline src/server/conn.rs]: any suggestion?

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Comment by hhkbp2** [comment]
Global metric maybe lead to performance reduction. A sysbench should be run to check that.
LGTM

**Pull request merged**